net/observation/handler.go

```diff
@@ -240,6 +251,9 @@ func (o *Observation[C]) wantBeNotified(r *pool.Message) bool {
 	if ValidSequenceNumber(o.private.obsSequence, obsSequence, o.private.lastEvent, now) {
 		o.private.obsSequence = obsSequence
 		o.private.lastEvent = now
+		if etag, err := r.ETag(); err == nil {
+			o.private.etag = etag
```

Comment:
use copy instead of assignation -> the underlayer buffer could be reused.

Reply:
copy() copies works based on length not capacity, so if the lengths do not match I have to allocate a new slice